### PR TITLE
Fix constant simd vector code in C 

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/2794

### CODE REVIEW

File: tests/reference/c-pragma2-a14de52.stdout [hunk 1]
```diff
@@ -27,6 +27,6 @@ int main(int argc, char* argv[])
 {
     _lpython_set_argv(argc, argv);
     float a __attribute__ (( vector_size(sizeof(float) * 8) ));
-    a = {(float)(1), (float)(1), (float)(1), (float)(1), (float)(1), (float)(1), (float)(1), (float)(1)};
+    a = (float __attribute__ (( vector_size(sizeof(float) * 8) ))) {(float)(1), (float)(1), (float)(1), (float)(1), (float)(1), (float)(1), (float)(1), (float)(1)};
```
Comment (Shaikh-Ubaid) [inline tests/reference/c-pragma2-a14de52.stdout:30]: It generates code as expected in https://github.com/lfortran/lfortran/issues/2794#issue-1975225866.

File: tests/reference/pass_where-where_03-00685f9.stdout [hunk 2]
```diff
@@ -456,7 +456,7 @@
                                                     (Real 4)
                                                 )]
                                                 (Array
-                                                    (Integer 4)
+                                                    (Real 4)
```
Comment (certik) [inline tests/reference/pass_where-where_03-00685f9.stdout:459]: Are these changes correct?

**Review by certik — APPROVED** [review]
Looks great overall, thanks!

I left a question.

File: tests/reference/pass_where-where_03-00685f9.stdout (hunk 2 above)
Comment (Thirumalai-Shaktivel) [reply tests/reference/pass_where-where_03-00685f9.stdout:459]: I verified some, it seems to be correct.

Comment (Shaikh-Ubaid) [reply tests/reference/pass_where-where_03-00685f9.stdout:459]: Same here. I verified couple of them before submitting here.

**Comment by Shaikh-Ubaid** [comment]
Thanks for the review and approval!

**Pull request merged**